AsyncDisplayKit/Details/ASDataController.mm

```diff
@@ -140,6 +140,11 @@ - (void)batchLayoutNodesFromContexts:(NSArray<ASIndexedNodeContext *> *)contexts
 #if AS_MEASURE_AVOIDED_DATACONTROLLER_WORK
     [ASDataController _expectToInsertNodes:contexts.count];
 #endif
+  
+  if (contexts.count == 0) {
+    batchCompletionHandler(@[], @[]);
```

Comment:
Is a comment justified for this one?  I assume this is like zero insertions or deletions, and ultimately calls back to...which code?

If you'd rather the reader follow the code, I trust your judgement.  Just a bit mysterious in a diff review that it needs an early return handler.